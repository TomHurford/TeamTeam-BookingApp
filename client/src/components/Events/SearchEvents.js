import React from "react";
const axios = require('axios').default;
import Event from "./Event";
import "../../styles/TitleOfPage.css"
import "../../styles/Home.css"

class SearchEvents extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      query: "",
      results: []
    }; 
    this.fetchDataOnLoad(); 
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({ query: event.target.value });
  }

  fetchDataOnLoad = async () => {
    const searchParams = new URLSearchParams(window.location.search);
    const eventName = await String(searchParams.get('name'));
    this.setState({query: eventName})
    const events = await this.getEventsByName(eventName);
    this.setState({results: events})
  }

getEventsByName  = async (eventName) => {
    console.log("getEventsByName");
    console.log(eventName);
    if(this.state.query === "" && eventName === undefined){
      console.log("getEventsByName Outer if");
      return [];
    }
    else{
      var event_name;
      if(eventName !== undefined){
        console.log("getEventsByName if");
        event_name = eventName;
      }
      else{
        console.log("getEventsByName else");
        event_name = this.state.query;
      }
      try{
        console.log("getEventsByName try");
        console.log(this.state.query);
        const response = await axios.post('http://localhost:5001/events/search', {name: event_name});
        const event = await response.data;
        const events = await event.event;
        console.log(await event);
        return events;
      } catch (error) {
        return null;
      }
    }
  }

async fetchData() {
    const events = await this.getEventsByName();
    this.setState({results: events})
  } 


  render() {
    return (
      <div className="title">
          <input
            className="searchBar"
            type="text"
            placeholder="Search for events..."
            value={this.state.query}
            onChange={this.handleChange}
            onKeyDown = {(e) => { if (e.key === 'Enter') { this.newSearch(this.state.query) }}}
          />

          <div className="events" data-testid="events-list">
            {this.state.results.map(event => (    
                <div className="eventCard" key={event.id} onClick={()=>this.handleClick(event.id)} >
                <Event details={event.id} specificEvent = {event}/>
                </div>
            ))}
          </div>
      </div>
    );
  }

  handleClick= (eventId) => {
    window.location.href = '/event-details?eventId=' + eventId;
}

 newSearch = (eventName) => {
    window.location.href = '/search-events?name=' + eventName;
  }
}

export default SearchEvents;