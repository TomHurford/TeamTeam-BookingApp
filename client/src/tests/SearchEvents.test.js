import renderer from 'react-test-renderer';
import React from 'react';
import SearchEvents from '../components/Events/SearchEvents';
// Snapshot test for searching events
test('SearchEvents snapshot test', () => {
  const component = renderer.create(<SearchEvents />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});