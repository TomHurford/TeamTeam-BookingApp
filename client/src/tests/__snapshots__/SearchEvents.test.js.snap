// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchEvents snapshot test 1`] = `
<div
  className="page-container"
>
  <div
    className="underlay"
  />
  <div
    className="title"
  >
    <input
      className="searchBar"
      data-testid="search-bar"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for events..."
      type="text"
      value=""
    />
    <div
      className="events"
      data-testid="events-list"
    />
  </div>
</div>
`;
